### PR TITLE
Add new extensions of `DispatchQueue`

### PR DESCRIPTION
**Pull request by jianstm**
Add `isMainQueue` to check if current queue is main queue, add `isCurrent(_:)` to check if current queue is specified queue. It's very useful to avoid `sync` on a serial queue. eg:

```Swift
if DispatchQueue.isMainQueue { 
    return
}

let logQueue = DispatchQueue(label: "logQueue")
if DispatchQueue.isCurrent(logQueue) {
    return 
}
```

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/Extensions/Dispatch/DispatchQueueExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#if canImport(Dispatch)
+import Dispatch
+
+// MARK: - Properties
+public extension DispatchQueue {
+
+    /// SwifterSwift: A Boolean value indicating whether the current
+    /// dispatch queue is the main queue.
+    public static var isMainQueue: Bool {
+        enum Static {
```
Comment (guykogus) [inline Sources/Extensions/Dispatch/DispatchQueueExtensions.swift]: Can this be marked as `private`? If so you should probably add it.

Comment (jianstm) [reply Sources/Extensions/Dispatch/DispatchQueueExtensions.swift]: No, it can't be `private`, will get a compile error

**Comment by guykogus** [comment]
@jianstm `pod lib lint` is failing. Can you please check locally to see what's wrong?
You also have some merge conflicts to resolve now.

**Comment by jianstm** [comment]
Hi @guykogus , I ran `pod lib lint` locally and got `SwifterSwift passed validation`. 
The latest ci checks said `The command "bundle exec danger --verbose" exited with 1.` and `The command "pod lib lint" exited with 0.` I think maybe there are something wrong with travis.

**Comment by pvtmert** [comment]
failed test related to urlextensions, https://travis-ci.org/SwifterSwift/SwifterSwift/builds/441282161#L1772

**Comment by guykogus** [comment]
Nice catch @pvtmert.
@jianstm can you please double-check that the iOS tests run locally? I don't see any reason why they shouldn't. If they're running correctly it may just be a little hiccup in travis, in which case if you can make an empty git commit like so:
```
git commit --amend
git push -f
```
That should trigger travis to run again.

File: Sources/Extensions/Dispatch/DispatchQueueExtensions.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (jianstm) [reply Sources/Extensions/Dispatch/DispatchQueueExtensions.swift]: Hi, @LucianoPAlmeida , it's just my personal preferences. They have the same purpose, but using `struct` means you can create an instance of it, like `let s = Static()`, using `enum` will not have such concern.

[the remark this answers is not in the document]
Comment (LucianoPAlmeida) [reply Sources/Extensions/Dispatch/DispatchQueueExtensions.swift]: I see, I just found caseless enum weird 😆 
But no problem 👍

**Pull request merged**